src/main/java/org/reaktivity/command/log/internal/LogCountersCommand.java

```diff
@@ -69,28 +72,31 @@ private void onDiscovered(
     private void counters(
         Path controlPath)
     {
-        try (ControlLayout layout = new ControlLayout.Builder()
-                .controlPath(controlPath)
-                .commandBufferCapacity(commandBufferCapacity)
-                .responseBufferCapacity(responseBufferCapacity)
-                .counterLabelsBufferCapacity(counterLabelsBufferCapacity)
-                .counterValuesBufferCapacity(counterValuesBufferCapacity)
-                .readonly(true)
-                .build())
+        if(!pathControlLayout.containsKey(controlPath))
         {
-            String owner = controlPath.getName(controlPath.getNameCount() - 2).toString();
-            CountersManager manager = new CountersManager(layout.counterLabelsBuffer(), layout.counterValuesBuffer());
-            manager.forEach((id, name) -> out.printf("%s.%s %d\n", owner, name, manager.getCounterValue(id)));
+            pathControlLayout.put(controlPath, new ControlLayout.Builder()
+                    .controlPath(controlPath)
+                    .commandBufferCapacity(commandBufferCapacity)
+                    .responseBufferCapacity(responseBufferCapacity)
+                    .counterLabelsBufferCapacity(counterLabelsBufferCapacity)
+                    .counterValuesBufferCapacity(counterValuesBufferCapacity)
+                    .readonly(true)
+                    .build());
         }
+        ControlLayout layout = pathControlLayout.get(controlPath);
+        String owner = controlPath.getName(controlPath.getNameCount() - 2).toString();
+        CountersManager manager = new CountersManager(layout.counterLabelsBuffer(), layout.counterValuesBuffer());
```

Comment:
Rather then store the layout in the map, why not store the CountersManager to eliminate the object creation (also, I'm not sure if the nmap is done in the CountersManager or the layout).